### PR TITLE
Upgrade .NET SDK to 6.0.100

### LINKED ISSUE #41: Update to net6.0
We should probably remove net5.0 support in favor of net6.0 LTS.
This will also allow us to use new language features and make the code a bit neater.

@jetersen Any objections to this?

### PR DESCRIPTION
**Pull request by jetersen**
fixes #41 

Not sure if I should include file scoped namespace and global using in this PR?

@patriksvensson also might not be ready yet due to Microsoft.Win32.Registry being preview.

### CODE REVIEW

**Comment by jetersen** [comment]
@patriksvensson WDYT
Is it good to ship or should we wait for Microsoft.Win32.Registry non preview?

Created https://github.com/dotnet/runtime/issues/62009 to get clarification on what is expected.

**Review by patriksvensson — CHANGES_REQUESTED** [review]
Looks good to me!

File: src/Cupboard.Providers.Windows/Cupboard.Providers.Windows.csproj [hunk 1]
```diff
@@ -12,10 +11,9 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Management.Infrastructure" Version="2.0.0" />
-    <PackageReference Include="Microsoft.Win32.Registry" Version="5.0.0" />
+    <PackageReference Include="System.Management" Version="6.0.0" />
```
[the remark this answers is not in the document]
Comment (patriksvensson) [reply src/Cupboard.Providers.Windows/Cupboard.Providers.Windows.csproj:14]: That's really good!

**Comment by patriksvensson** [comment]
Merged! Thank you for your contribution. Much appreciated! 👍

File: src/Cupboard.Providers.Windows/WmiFacts.cs [hunk 2]
```diff
@@ -31,18 +31,18 @@ public IEnumerable<(string Name, object Value)> GetFacts(IRemainingArguments arg
...
+            var keys = string.Join(',', _properties.Keys);
+            var searcher = new ManagementObjectSearcher($"SELECT {keys} FROM Win32_OperatingSystem");
+            var osDetailsCollection = searcher.Get();
+            foreach (var prop in osDetailsCollection)
             {
-                if (_properties.ContainsKey(prop.Name))
+                foreach (var property in _properties)
                 {
-                    yield return (_properties[prop.Name], prop.Value);
+                    yield return (property.Value, prop[property.Key]);
```
Comment (jetersen) [inline src/Cupboard.Providers.Windows/WmiFacts.cs:41]: FYI, this for each is necessary to get the enumeration going but there should only be returned one object.
These are not the nested for each loop your looking for.

**Pull request merged**